src/Orleans.Runtime/Core/ManagementGrain.cs

```diff
@@ -208,6 +212,25 @@ public Task<object[]> SendControlCommandToProvider(string providerTypeFullName,
                 String.Format("SendControlCommandToProvider of type {0} and name {1} command {2}.", providerTypeFullName, providerName, command));
         }
 
+        public ValueTask<SiloAddress> GetActivationAddress(IAddressable reference)
```

Comment:
Unless I am mistaken, this will not work for `StatelessWorker`. Maybe that's fine, but I think we should throw or return an error in this case

Reply:
Oh! I forgot them. Will test and see.

Reply:
@benjaminpetit I've added a comment regarding the workers returning `null`. It doesn't throw when doing the lookup nor I've found an easy way to detect if the grain is a worker without using reflection on the type to find the attribute.

Reply:
It's doable with the `IClusterManifestProvider`. You should try to find the properties for the `GrainType` (it's in `GrainId`), and withing these properties check the values of the one with the
`WellKnownGrainTypeProperties.PlacementStrategy` key

Reply:
Ok, done! 👍🏻